include/igl/copyleft/marching_cubes.h

```diff
@@ -47,7 +47,36 @@ namespace igl
         const unsigned z_res,
         Eigen::PlainObjectBase<DerivedV2> &vertices,
         Eigen::PlainObjectBase<DerivedF> &faces);
+
+      // marching_cubes( values, points, indices, vertices, faces )
+      //
+      // Perform marching cubes reconstruction on the grid cells defined by indices.
+      // The indices parameter is an nx8 dense array of index values into the points and values arrays.
+      // Each row of indices represents a cube for which to generate vertices and faces over.
+      //
+      // Input:
+      //  values  #number_of_grid_points x 1 array -- the scalar values of an
+      //    implicit function defined on the grid points (<0 in the inside of the
+      //    surface, 0 on the border, >0 outside)
+      //  points  #number_of_grid_points x 3 array -- 3-D positions of the grid
+      //    points, ordered in x,y,z order:
+      //  indices  #cubes x 8 array -- one row for each cube where each value is
+      //    the index of a vertex in points and a scalar in values.
+      //    i.e. points[indices[i, j]] = the position of the j'th vertex of the i'th cube
+      // Output:
+      //   vertices  #V by 3 list of mesh vertex positions
+      //   faces  #F by 3 list of mesh triangle indices
+      // Note: The winding direction of the cube indices will affect the output winding of the faces
+      //
+      template <typename DerivedV1, typename DerivedV2, typename DerivedI, typename DerivedF>
+      IGL_INLINE void marching_cubes(
+        const Eigen::PlainObjectBase<DerivedV1> &values,
```

Comment:
Input types should be templated from Eigen::MatrixBase (this is a new guideline and we're slowly changing old functions).